rmw_connext_shared_cpp/src/node.cpp

```diff
@@ -34,13 +37,42 @@ create_node(
   const char * name,
   const char * namespace_)
 {
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(context, NULL);
+  assert(implementation_identifier != NULL);
```

Comment:
This one seems like one where we should `RMW_CHECK_ARGUMENT_FOR_NULL`.  At the very least, that will make it operate the same as the checks for other NULL below, like `context`.

Also, here and elsewhere, we can use `nullptr` instead of `NULL`.

Reply:
> This one seems like one where we should RMW_CHECK_ARGUMENT_FOR_NULL. At the very least, that will make it operate the same as the checks for other NULL below, like context.

We could, but given that this is a somewhat internal API (not a standalone RMW implementation) I'd rather not. But I give you that an `assert` here is unwarranted and almost redundant with the check below. Removed it.